src/java.base/share/classes/java/lang/reflect/code/op/CoreOps.java

```diff
@@ -1599,7 +1599,7 @@ static List<Value> operands(Value array, Value index, Value v) {
         }
 
         static TypeElement resultType(Value array, Value v) {
-            JavaType arrayType = (JavaType) array.type();
+            ArrayType arrayType = (ArrayType) array.type();
             if (!arrayType.isArray()) {
                 throw new IllegalArgumentException("Type is not an array type: " + arrayType);
             }
```

Comment:
```suggestion
            if (!(array.type() instanceof ArrayType arrayType)) {
                throw new IllegalArgumentException("Type is not an array type: " + array.type());
            }
```
?

Reply:
How do I build the examples, and make sure everything is ok?

Reply:
Ok, I found the POM files. Note: the SPIR-V project doesn't build, as the POM depends on version 0.3 of `beehive-spirv-toolkit`, but when installing that package locally, the build ends up with 0.4 (I guess the source has been updated in the meantime).

Reply:
Ok, i will look at that separately.